src/pyodide/internal/workers.py

```diff
@@ -149,54 +157,204 @@ def json(
             raise _to_python_exception(exc) from exc
 
 
-# TODO: Implement pythonic blob API
-FormDataValue = "str | js.Blob"
+FormDataValue = "str | js.Blob | Blob"
 
 
 class FormData(MutableMapping[str, FormDataValue]):
-    def __init__(self, form_data: "js.FormData | None | dict[str, FormDataValue]"):
+    """
+    This API follows that of https://pypi.org/project/multidict/.
+    """
+
+    def __init__(
+        self, form_data: "js.FormData | None | dict[str, FormDataValue]" = None
+    ):
```

Comment:
I think it's nicer style to do something like:
```
if not form_data:
   self._js_form_data = js.FormData.new()
   return 
if hasattr(form_data, "constructor") and form_data.constructor.name == "FormData":
   self._js_form_data = form_data
   return
if isinstance(form_data, dict):
   self._js_form_data = _dict_to_from_data(form_data)
   return
raise TypeError("Expected form_data to be a dict or an instance of FormData")
```
and then factor out the logic to convert a dictionary to a from data into a separate function.

Reply:
Agreed